### PR TITLE
Add Minkowski

### PR DESCRIPTION
**Pull request by osheamonn**

### CODE REVIEW

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/MakeCreatableFromYaml.hpp"
+#include "Options/Options.hpp"
+
+namespace EinsteinSolutions {
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: don't do what Larry does, move the definitions out of the header file

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+/*!
+ * \ingroup EinsteinSolutions
+ * \brief The Minkowski solution for flat space in Dim spatial dimensions.
+ *
+ * \details Flat space has lapse \f$N(x,t)= 1 \f$, shift \f$N^i(x,t) = 0
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: check the doxygen (math broken across lines)

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+#include "DataStructures/MakeWithValue.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/MakeCreatableFromYaml.hpp"
+#include "Options/Options.hpp"
+
+namespace EinsteinSolutions {
+
+/*!
+ * \ingroup EinsteinSolutions
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: did you define this?

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp (hunk 2 above)
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: Flat space -> The solution has

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+  Minkowski() = default;
+  Minkowski(const Minkowski& /*rhs*/) = delete;
+  Minkowski& operator=(const Minkowski& /*rhs*/) = delete;
+  Minkowski(Minkowski&& /*rhs*/) noexcept = default;
+  Minkowski& operator=(Minkowski&& /*rhs*/) noexcept = default;
+  ~Minkowski() = default;
+
+  template <typename T>
+  Scalar<T> lapse(const tnsr::I<T, Dim>& x, double /*t*/) const noexcept;
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: try /*t*/ -> t

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+  template <typename T>
+  tnsr::II<T, Dim> inverse_spatial_metric(const tnsr::I<T, Dim>& x, double /*t*/
+                                          ) const noexcept;
+  template <typename T>
+  tnsr::ii<T, Dim> extrinsic_curvature(const tnsr::I<T, Dim>& x,
+                                       double /*t*/) const noexcept;
+
+  // clang-tidy: google-runtime-references
+  void pup(PUP::er& /*p*/){};  // NOLINT
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: noexcept and no ;

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+  return tnsr::iJ<T, Dim>(make_with_value<T>(x, 0.));
+}
+
+template <size_t Dim>
+template <typename T>
+tnsr::ii<T, Dim> Minkowski<Dim>::spatial_metric(const tnsr::I<T, Dim>& x,
+                                                const double /*t*/) const
+    noexcept {
+  tnsr::ii<T, Dim> lower_metric(make_with_value<T>(x, 0.));
+  for (size_t i = 0.; i < Dim; ++i) {
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: 0;

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+    lower_metric.get(i, i) = make_with_value<T>(x, 1.);
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: don't need make_with_value

File: src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+template <size_t Dim>
+template <typename T>
+tnsr::II<T, Dim> Minkowski<Dim>::inverse_spatial_metric(
+    const tnsr::I<T, Dim>& x, const double /*t*/
+    ) const noexcept {
+  tnsr::II<T, Dim> upper_metric(make_with_value<T>(x, 0.));
+  for (size_t i = 0; i < Dim; ++i) {
+    upper_metric.get(i, i) = make_with_value<T>(x, 1.);
```
Comment (kidder) [inline src/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Minkowski.hpp]: ditto

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_Minkowski.cpp [hunk 9]
```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <limits>
```
Comment (kidder) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_Minkowski.cpp]: remove

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_Minkowski.cpp [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+    for (size_t j = 0; j < Dim; ++j) {
+      CHECK(dt_g.get(i, j) == zero);
+      CHECK(extrinsic_curvature.get(i, j) == zero);
+      for (size_t k = 0; k < Dim; ++k) {
+        CHECK(d_g.get(i, j, k) == zero);
+      }
+    }
+  }
+}
+
```
Comment (kidder) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/Test_Minkowski.cpp]: PUP RIP (just call serialize_and_deserialize)

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Once #338 is merged all the member functions can be moved to a cpp file and have explicit instantiations generated similar to the way ScalarWave::Solutions::PlaneWave is done in #338

**Review by kidder — DISMISSED** [review]
squash

File: docs/GroupDefs.hpp [hunk 11]
```diff
@@ -76,6 +76,11 @@
  */
 
 /*!
+ * \defgroup EinsteinSolutions Einstein Solutions
```
Comment (nilsdeppe) [inline docs/GroupDefs.hpp]: Please change the group name to `EinsteinSolutionsGroup` to avoid collision with namespace name.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Uh oh

**Review by nilsdeppe — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**